framework/src/main/java/org/checkerframework/framework/util/FlowExpressionParseUtil.java

```diff
@@ -181,12 +257,23 @@ private static boolean isMemberSelect(String s, FlowExpressionContext context) {
                     array.first.first + "[" + array.first.second + "]", array.second.substring(1));
         }
 
-        Pattern memberSelectOfStringPattern = anchored("(" + STRING_REGEX + ")" + "\\.(.*)");
-        Matcher m = memberSelectOfStringPattern.matcher(s);
-        if (m.matches()) {
-            return Pair.of(m.group(1), m.group(2));
+        // parses member select whose receiver is a string literal
+        try {
+            Expression e = parseExpression(s);
+            if (e.getClass().equals(MethodCallExpr.class)) {
+                MethodCallExpr methodCallExpr = (MethodCallExpr) e;
+                String mceString = methodCallExpr.toString();
+                String receiver = methodCallExpr.getScope().get().toString();
+                String remaining = mceString.substring(receiver.length(), mceString.length());
+                if (receiver.charAt(0) == '\"') {
+                    return Pair.of(receiver, remaining.substring(1));
+                }
+            }
+        } catch (ParseProblemException pe) {
+            // if format of the expression is such that it cannot be parsed by JavaParser
```

Comment:
This is not grammatical (it starts with "if" but doesn't have a consequent).  It also doesn't explain why the exception is caught but ignored.